nixos/modules/services/network-filesystems/kbfs.nix

```diff
@@ -0,0 +1,80 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  cfg = config.services.kbfs;
+
+in {
+
+  ###### interface
+
+  options = {
+
+    services.kbfs = {
+
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Whether to mount the Keybase filesystem.";
+      };
+
+      mountPoint = mkOption {
+        type = types.str;
+        default = "/keybase";
+        example = "/home/user/keybase";
+        description = "Mountpoint for the Keybase filesystem.";
+      };
+
+      createMountPoint = mkOption {
+        type = types.bool;
+        default = true;
+        description = ''
+          Whether to create the mount point directory at startup time.
+        '';
+      };
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.kbfs;
+        defaultText = "pkgs.kbfs";
+        example = literalExample "pkgs.kbfs";
+        description = "Keybase filesystem derivation to use.";
+      };
+
+      extraFlags = mkOption {
+        type = types.listOf types.str;
+        default = [];
+        example = [
+          "-label kbfs"
+          "-mount-type normal"
+        ];
+        description = ''
+          Additional flags to pass to the Keybase filesystem on launch.
+        '';
+      };
+
+    };
+  };
+
+  ###### implementation
+
+  config = mkIf cfg.enable {
+
+    systemd.user.services.kbfs = {
+      description = "Keybase File System";
+      requires = [ "keybase.service" ];
```

Comment:
We should also have ```after = [ "keybase.service" ];``` although kbfs WILL still throw an error in the beginning because the keybase service isn't up.